src/coreclr/vm/interpexec.cpp

```diff
@@ -2117,15 +2152,24 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     callArgsOffset = ip[2];
                     int32_t calliFunctionPointerVar = ip[3];
                     int32_t calliCookie = ip[4];
+                    int32_t flags = ip[5];
 
                     void* cookie = pMethod->pDataItems[calliCookie];
-                    ip += 5;
+                    ip += 6;
 
                     // Save current execution state for when we return from called method
                     pFrame->ip = ip;
 
                     // Interpreter-FIXME: isTailcall
-                    InvokeCalliStub(LOCAL_VAR(calliFunctionPointerVar, PCODE), cookie, stack + callArgsOffset, stack + returnOffset);
+                    if ((flags & (int32_t)CalliFlags::PInvoke) && !(flags & (int32_t)CalliFlags::SuppressGCTransition) && !(flags & (int32_t)CalliFlags::PInvokeMarshalled))
+                    {
+                        InvokePInvokeCalliStub(LOCAL_VAR(calliFunctionPointerVar, PCODE), cookie, stack, pFrame, stack + callArgsOffset, stack + returnOffset);
```

Comment:
It would be nice to have a clear separation of managed and unmanaged paths, similar to what we do for direct method invocations.

- InvokeUnmanagedCalliWithTransition (equivalent of InvokeUnmanagedMethodWithTransition)
- InvokeUnmanagedCalli (equivalent of InvokeUnmanagedMethod)
- InvokeManagedCalli (equivalent of InvokeManagedMethod)